src/main/scala/d2dadapter/CRCGenerator.scala

```diff
@@ -0,0 +1,132 @@
+package edu.berkeley.cs.ucie.digital.d2dadapter
+
+import chisel3._
+import chisel3.util._
+import chisel3.util.Decoupled
+import edu.berkeley.cs.ucie.digital.d2dadapter.CRC16Lookup
+
+/** Generates the CRC of data using the polynomial x^16 + x^15 + x^2 + 1. The
+  * message must be formatted according to the UCIe1.1 spec, and the message
+  * will be consumed one byte per clock cycle, starting from the MSB to the LSB.
+  * The calculated 2 byte CRC will be available on the crc ReadyValid3IO
+  * interface in the raw, un-reversed format as indicated in the UCIe1.1 spec
+  * (i.e. CRC[15] is the MSB of the crc interface's data bits).
+  * @param width
+  *   The message size in bits (must be whole number of bytes). UCIe1.1 spec
+  *   will use 1024-bit messages, with shorter messages padded with 0s in MSB
+  *   side.
+  * @groupdesc Signals
+  *   The input and output controls of the CRCGenerator module
+  */
+
+class CRCGenerator(width: Int) extends Module { //
+  val io = IO(new Bundle {
+
+    /** Resets the CRCGenerator, and sets the default message ready state high
+      * and crc valid state low. CRC bits will be cleared to 0.
+      * @group Signals
+      */
+    val reset = Input(Bool())
```

Comment:
use implicit reset instead

Reply:
Would I just be able to omit the reset declaration in the IO. I also changed it so that when a new message is latched, the crc output is cleared and set invalid, let me know if this is correct though.